### PR TITLE
feat(ext-links): Adding external icon to links based on a parameter -  FRONT-3601

### PR DESCRIPTION
**Pull request by planctus**
The approach used is to give priority to the external icon:
it means that if the link is set to be external it will take precedence on any other "icon" that might have been passed to the component.
This can be tested by setting an icon through the controls, if the link is external nothing should change.

### CODE REVIEW

File: src/specs/components/card/demo/data--card.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ module.exports = {
       label: 'Better regulation',
       path: exampleLink,
       level: 1,
+      external: true,
```
Comment (emeryro) [inline src/specs/components/card/demo/data--card.js]: for the demo to COMM we can have the external link by default, but I guess that on the end we should have the normal link here (and maybe a control to have it external)

File: src/specs/components/content-item/demo/data--image.js [hunk 2]
```diff
@@ -16,6 +16,8 @@ module.exports = {
     type: 'standalone',
     label: 'Title',
     path: exampleLink,
+    icon_path: 'icons.svg',
+    external: true,
```
Comment (emeryro) [inline src/specs/components/content-item/demo/data--image.js]: same remark for the default external link

File: src/specs/components/menu/demo/data--en.js [hunk 3]
```diff
@@ -36,25 +36,7 @@ module.exports = {
...
-        { label: 'Item 4.7', path: exampleLink },
-        { label: 'Item 4.8', path: exampleLink },
-        { label: 'Item 4.9', path: exampleLink },
-        { label: 'Item 4.10', path: exampleLink },
-        { label: 'Item 4.11', path: exampleLink },
-        { label: 'Item 4.12', path: exampleLink },
-        { label: 'Item 4.13', path: exampleLink },
-        { label: 'Item 4.14', path: exampleLink },
-      ],
+      external: true,
```
Comment (emeryro) [inline src/specs/components/menu/demo/data--en.js]: we should update the translated data file too

**Review by emeryro — CHANGES_REQUESTED** [review]

**Comment by planctus** [comment]
Thanks Romain for reviewing this, the examples were meant to show where the "functionality" was added to give a chance both to comm and platform to see what the result would have been. I agree that if this gets "validated" by our stakeholders we can simply remove the icon in some of the stories where we are currently showing it, i would also keep the control to activate it only for the link stories.

**Comment by planctus** [comment]
It still not entirely clear how are we going to release this, the pr has been kept against v3, the examples have been removed, the link is the main "entry" point to test the "external icon" using the associated control

**Comment by planctus** [comment]
The external icon has been added also to the tag component, only when the tag is of type "link"

**Review by emeryro — APPROVED** [review]

**Pull request merged**